File: frontend/src/constants/techStack.ts
export interface TechOption {
  value: string;
  label: string;
}

export const FRONTEND_OPTIONS: TechOption[] = [
  { value: "", label: "Select frontend..." },
  { value: "html_css_js", label: "HTML / CSS / JavaScript" },
  { value: "react", label: "React" },
  { value: "vue", label: "Vue.js" },
  { value: "angular", label: "Angular" },
  { value: "nextjs", label: "Next.js" },
  { value: "svelte", label: "Svelte" },
];

export const BACKEND_OPTIONS: TechOption[] = [
  { value: "", label: "Select backend..." },
  { value: "node_express", label: "Node.js (Express)" },
  { value: "python_fastapi", label: "Python (FastAPI)" },
  { value: "python_django", label: "Python (Django)" },
  { value: "java_spring", label: "Java (Spring Boot)" },
  { value: "dotnet", label: ".NET" },
  { value: "go", label: "Go" },
];

export const DATABASE_OPTIONS: TechOption[] = [
  { value: "", label: "Select database..." },
  { value: "postgresql", label: "PostgreSQL" },
  { value: "mysql", label: "MySQL" },
  { value: "mongodb", label: "MongoDB" },
  { value: "sqlite", label: "SQLite" },
  { value: "redis", label: "Redis" },
];

export function techLabel(
  value: string | null | undefined,
  options: TechOption[]
): string {
  if (!value) return "—";
  return options.find((o) => o.value === value)?.label ?? value;
}

File: backend/app/services/tech_stack.py
TECH_LABELS: dict[str, str] = {
    "html_css_js": "HTML / CSS / JavaScript",
    "react": "React",
    "vue": "Vue.js",
    "angular": "Angular",
    "nextjs": "Next.js",
    "svelte": "Svelte",
    "node_express": "Node.js (Express)",
    "python_fastapi": "Python (FastAPI)",
    "python_django": "Python (Django)",
    "java_spring": "Java (Spring Boot)",
    "dotnet": ".NET",
    "go": "Go",
    "postgresql": "PostgreSQL",
    "mysql": "MySQL",
    "mongodb": "MongoDB",
    "sqlite": "SQLite",
    "redis": "Redis",
}


def tech_label(value: str | None) -> str | None:
    if not value:
        return None
    return TECH_LABELS.get(value, value)


def compose_known_technologies(
    frontend: str | None,
    backend: str | None,
    database: str | None,
) -> str:
    parts: list[str] = []
    if frontend:
        parts.append(f"Frontend: {tech_label(frontend)}")
    if backend:
        parts.append(f"Backend: {tech_label(backend)}")
    if database:
        parts.append(f"Database: {tech_label(database)}")
    return " | ".join(parts)
